aten/src/ATen/native/quantized/cpu/qpool.cpp

```diff
@@ -244,12 +247,141 @@ Tensor quantized_max_pool2d(
 namespace {
 class QMaxPool2D_arr_args final : public torch::OperatorKernel {
  public:
+  #ifdef USE_PYTORCH_QNNPACK
+   Tensor qnnpack_maxpool(
+       Tensor input,
+       IntArrayRef kernel_size,
+       IntArrayRef stride,
+       IntArrayRef padding,
+       IntArrayRef dilation) {
+     Tensor qy;
+
+     TORCH_CHECK(
+         input.ndimension() == 4,
+         "qnnpack_maxpool(): Expected input to be 4-dimensional: got ",
+         input.ndimension());
+     TORCH_CHECK(
+         kernel_size.size() == 2,
+         "qnnpack_maxpool(): Expected kernel_size to be 2-dimensional: got ",
+         kernel_size.size());
+     TORCH_CHECK(
+         stride.size() == 2,
+         "qnnpack_maxpool(): Expected stride to be 2-dimensional: got ",
+         stride.size());
+     TORCH_CHECK(
+         dilation.size() == 2,
+         "qnnpack_maxpool(): Expected dilation to be 2-dimensional: got ",
+         dilation.size());
+     TORCH_CHECK(
+         padding.size() == 2,
+         "qnnpack_maxpool(): Expected padding to be 2-dimensional: got ",
+         padding.size());
+
+     Tensor input_contig = input.contiguous();
```

Comment:
Do you need a check for input layout here? How do you know that the input is nhwc contiguous?

Reply:
I'll make the change similar to https://github.com/pytorch/pytorch/pull/26242 once that lands for all qnnpack ops.